src/helpers/slack.ts

```diff
@@ -0,0 +1,109 @@
+const { Reaction, ErrorCode } = require('@slack/web-api');
+import SLACK_MESSAGE_BLOCKS from './files/welcomeMessageBlocks.json';
+import { SlackEvent, SlackWebClient } from './types';
+const fs = require('fs');
+
+
+function getEmojisToReactWith(text: string): Array<string> {
+    const emojisToReactWith: Array<string> = [];
+
+    let rawdata = fs.readFileSync('./helpers/files/emojis.json');
+    let emojis: Record<string, Array<string>> = JSON.parse(rawdata);
+
+    for (const [emoji, keywords] of Object.entries(emojis)) {
+        for (const keyword of keywords) {
+            if (text.includes(keyword)) {
+                emojisToReactWith.push(emoji)
+            }
+        }
+    }
+
+    console.log(`Emojis to react with: ${emojisToReactWith}`)
+    return emojisToReactWith
+}
+
+async function getCurrentEmojisOnSlackPost(slackWebClient: SlackWebClient, channel: string, timestamp: string): Promise<string[]> {
+    try {
+        const reactions = await slackWebClient.reactions.get({
+            channel,
+            timestamp
+        });
+
+        const messageReactions = reactions?.message?.reactions;
+        if (messageReactions) {
+            // return an array of reactions
+            return messageReactions.map((reaction: typeof Reaction) => reaction.name);
+        } else {
+            console.log(`No reactions found for message with timestamp ${timestamp} in channel ${channel}.`);
+            return [];
+        }
```

Comment:
```suggestion
        if (!messageReactions) {
         console.log(`No reactions found for message with timestamp ${timestamp} in channel ${channel}.`);
             return [];
          
        } 
          // return an array of reactions
            return messageReactions.map((reaction: typeof Reaction) => reaction.name);
        
```

Else can be avoided like 90%+ of the cases. 🥇

Reply:
Nice once, updated